pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -293,11 +300,15 @@ func createBackupPod(ctx context.Context, nodeName, recentBackupName, operatorIm
 }
 
 // isRequireRecentBackup checks the conditions of ClusterVersion to verify if a backup is required.
-func isRequireRecentBackup(config *configv1.ClusterVersion) bool {
+func isRequireRecentBackup(config *configv1.ClusterVersion, clusterOperatorStatus *configv1.ClusterOperatorStatus) bool {
 	for _, condition := range config.Status.Conditions {
 		// Check if ReleaseAccepted is false and Message field containers the string RecentBackup.
 		if condition.Type == "ReleaseAccepted" && condition.Status == configv1.ConditionFalse {
-			return strings.Contains(condition.Message, backupConditionType)
+			if strings.Contains(condition.Message, backupConditionType) {
+				return true
+			} else if backupRequired, err := isNewBackupRequired(config, clusterOperatorStatus); err == nil && backupRequired {
```

Comment:
it might be quite late for me already, but while this is an easy addition the method name now doesn't matches the content anymore. Maybe you can add the check in the above `ensureRecentBackup`? There you can also pass the cluster version along with it (which also confused me why you had to parse it twice).

Reply:
haven't had a coffee yet, but that's just a logical OR between those? 

eg 
> return isRequireRecentBackup || isNewBackupRequired

Reply:
fixed :)